transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferSourceActivity.java

```diff
@@ -0,0 +1,116 @@
+
+import com.hyperwallet.android.ui.common.repository.Event;
+import com.hyperwallet.android.ui.common.view.ActivityUtils;
+import com.hyperwallet.android.ui.common.view.error.OnNetworkErrorCallback;
+import com.hyperwallet.android.ui.transfer.R;
+import com.hyperwallet.android.ui.transfer.TransferSourceWrapper;
+import com.hyperwallet.android.ui.transfer.viewmodel.ListTransferSourceViewModel;
+
+import java.util.ArrayList;
+
+public class ListTransferSourceActivity extends AppCompatActivity implements OnNetworkErrorCallback {
+
+    public static final String TAG = "transfer-funds:create-transfer";
+    public static final String EXTRA_SELECTED_SOURCE = "EXTRA_SELECTED_SOURCE";
+    public static final String EXTRA_SELECTED_SOURCE_TOKEN = "SELECTED_SOURCE_TOKEN";
+    public static final String EXTRA_TRANSFER_SOURCE_LIST = "TRANSFER_SOURCE_LIST";
+
+    private ListTransferSourceViewModel mListTransferSourceViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_transfer_source);
+
+        mListTransferSourceViewModel = ViewModelProviders.of(this,
+                new ListTransferSourceViewModel.ListTransferSourceViewModelFactory())
+                .get(ListTransferSourceViewModel.class);
+
+        String transferToken = getIntent().getStringExtra(EXTRA_SELECTED_SOURCE_TOKEN);
+        if (TextUtils.isEmpty(transferToken)) {
+            throw new IllegalArgumentException(
+                    "EXTRA_SELECTED_SOURCE_TOKEN intent data is needed to start this activity");
+        }
+
+        ArrayList<TransferSourceWrapper> sourceList = getIntent().getParcelableArrayListExtra(
+                EXTRA_TRANSFER_SOURCE_LIST);
+        if (savedInstanceState == null) {
+            ActivityUtils.initFragment(this, ListTransferSourceFragment.newInstance(transferToken, sourceList),
+                    R.id.list_source_fragment);
+        }
+
+        registerObservers();
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getWindow().setStatusBarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark));
+        getWindow().getDecorView().setSystemUiVisibility(0);
+        super.onBackPressed();
+    }
+
+    @Override
+    public void retry() {
+
+    }
+
+    @Override
+    protected void onStop() {
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getWindow().setStatusBarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark));
```

Comment:
some versions of android might not support `setStatusBarColor` and `setSystemUiVisibility`  or this will crash

Reply:
setStatusBarColor introduce from API level 21, App min Api level is 21 its not cause any crash. 
setSystemUiVisibility is deprecated from API level 30, still App max version is 28, setSystemUiVisibility commonly used in all activity.